domains/MatchRoom.js

```diff
@@ -150,6 +150,10 @@ class MatchRoom extends Room {
 
 			if (!is_replace_flow) {
 				this.sendStateToAdmin();
+			} else {
+				// TODO: if producer was assigned to user, shou;d we remove the user too?
+				// TODO: Question especially relevant for autojoin mode?
+				// NOTE: removing the player assignment, means that if the player has a connection blips, he might be kicked form the room and not have his game data resume as soon as he reconnects
```

Comment:
Removing disconnected users when autojoin is enabled can be benefical. However I would not directly trigger it from here. Check first if there is even another user, which is connecting/waiting for a slot. And only disconnect someone after a generous timeout counter passed, e.g. 30s.

Reply:
Agreed, smoothing out blips is a concern of the raw connection, or maybe the user. 

I've implemented a deferred leave room system in the PR, by 30s as you suggested. I'll do a quick test and I might just push that tonight.